packages/headless/src/controllers/url-manager/headless-url-manager.ts

```diff
@@ -0,0 +1,117 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Engine} from '../../app/headless-engine';
+import {restoreSearchParameters} from '../../features/search-parameters/search-parameter-actions';
+import {SearchParametersState} from '../../state/search-app-state';
+import {validateInitialState} from '../../utils/validate-payload';
+import {buildController, Controller} from '../controller/headless-controller';
+import {executeSearch} from '../../features/search/search-actions';
+import {ConfigurationSection} from '../../state/state-sections';
+import {buildSearchParameterSerializer} from '../../features/search-parameters/search-parameter-serializer';
+import {buildSearchParameterManager} from '../search-parameter-manager/headless-search-parameter-manager';
+import {logParametersChange} from '../../features/search-parameters/search-parameter-analytics-actions';
+import {initialSearchParameterSelector} from '../../features/search-parameters/search-parameter-selectors';
+
+export interface UrlManagerProps {
+  /**
+   * The initial state that should be applied to the `UrlManager` controller.
+   */
+  initialState: UrlManagerInitialState;
+}
+
+export interface UrlManagerInitialState {
+  /**
+   * The part of the url containing the parameters affecting the search response.
+   */
+  fragment: string;
+}
+
+const initialStateSchema = new Schema<Required<UrlManagerInitialState>>({
+  fragment: new StringValue(),
+});
+
+/**
+ * The `UrlManager` controller allows managing an url which affect the search response.
+ * */
+export interface UrlManager extends Controller {
+  /**
+   * The state relevant to the `UrlManager` controller.
+   * */
+  state: UrlManagerState;
+  /**
+   * Updates the search parameters from the url & launches a search.
```

Comment:
Although I can tell by looking at the code, with just this documentation line here it's not clear whether the search parameters in the state or in the url are the ones being updated. Suggestion:
```suggestion
   * Updates the search parameters in state with those from the url & launches a search.
```